### ISSUE TITLE
Modal IE10 ESC key does not initially close modal

### ISSUE THREAD

**Issue by Matt-Butler** [cerner/terra-core#335]
In IE10, the ESC key does not close the modal until the user has clicked on the modal. If you open the modal, click on it, then press esc it will still close. 

To reproduce the issue:
1) Use IE10 and navigate to any modal example that has closeOnEsc enabled.
2) Open the modal and push the ESC key

**Actual Behavior**
Modal doesn't close.

**Expected Behavior**
Modal closes.

**Comment by Matt-Butler** [comment]
![modal-ie10](https://cloud.githubusercontent.com/assets/8146159/25971134/d5fd4d6e-3660-11e7-86ef-0df01768a122.gif)

On IE10, it seems the focus is moving to the menu and no keyboard events are triggered.  I am also unable to tab into the modal

**Comment by Matt-Butler** [comment]
I think the issue is with initial focus
To reproduce the issue:
- navigate to http://engineering.cerner.com/terra-core/#/site/modal
- In the console, `document.activeElement`

Firefox:
```
document.activeElement
<div tabindex="0" aria-label="Terra Modal" class="terra-Modal" role="document">
```
Chrome:
```
document.activeElement
<body data-gr-c-s-loader="true">...</body>
```
IE10:
```
document.activeElement 
[object HTMLHtmlElement] {
	version : "",
	currentStyle : [object MSCurrentStyleCSSProperties],
	runtimeStyle : [object MSStyleCSSProperties],
	accessKey : "",
	classList : terra-Base,
	className : "terra-Base",
	contentEditable : "inherit",
	dir : "ltr",
	disabled : false,
	draggable : false
	...
} 
```
----

trap-focus-react has the option of specifying initialFocus: https://github.com/davidtheclark/focus-trap-react/blob/dd92067f32ffcbf12d02a0ae6646a219b1fd8cbb/test/activation.test.js#L55

I tried adding initialFocus as `.terra-Modal`
```
      <FocusTrap focusTrapOptions={{ initialFocus: '.terra-Modal' }}>
```

Firefox:
```
document.activeElement
<div tabindex="0" aria-label="Terra Modal" class="terra-Modal" role="document">
```
Chrome:
```
document.activeElement
<body data-gr-c-s-loader="true">...</body>
```
IE10:
```
document.activeElement 
[object HTMLHtmlElement] {
	version : "",
	currentStyle : [object MSCurrentStyleCSSProperties],
	runtimeStyle : [object MSStyleCSSProperties],
	accessKey : "",
	classList : terra-Base,
	className : "terra-Base",
	contentEditable : "inherit",
	dir : "ltr",
	disabled : false,
	draggable : false
	...
} 
Add to watch
```

**Comment by Matt-Butler** [comment]
I figured it out.

Adding this code inside the render function of the modal component fixed the issue.
```
    if (!isOpened) {
      return null
    }
```

**Comment by Matt-Butler** [comment]
https://github.com/cerner/terra-core/pull/344
